lib/ansible/modules/cloud/amazon/kms.py

```diff
@@ -0,0 +1,272 @@
+    ret = kms.list_keys()
+    for k in ret['Keys']:
+        if k['KeyId'] == key_id:
+            return k['KeyArn']
+    raise Exception('could not find key from id: {}'.format(key_id))
+
+def get_arn_from_role_name(iam, rolename):
+    ret = iam.get_role(RoleName=rolename)
+    if ret.get('Role') and ret['Role'].get('Arn'):
+        return ret['Role']['Arn']
+    raise Exception('could not find arn for name {}.'.format(rolename))
+
+def do_grant(kms, keyarn, role_arn, granttypes, mode='grant', dry_run=True):
+    ret = {}
+    keyret = kms.get_key_policy(KeyId=keyarn, PolicyName='default')
+    policy = json.loads(keyret['Policy'])
+
+    changes_needed = {}
+    assert_policy_shape(policy)
+    for statement in policy['Statement']:
+        for granttype in ['role', 'role grant', 'admin']:
+            # do we want this grant type? Are we on its statement?
+            # and does the role have this grant type?
+
+            if mode == 'grant' and statement['Sid'] == statement_label[granttype]:
+                # we're granting and we recognize this statement ID.
+
+                if granttype in granttypes:
+                    if not role_arn in statement['Principal']['AWS']: # needs to be added.
+                        changes_needed[granttype] = 'add'
+                        if not dry_run:
+                            statement['Principal']['AWS'].append(role_arn)
+                elif role_arn in statement['Principal']['AWS']: # not one the places the role should be
+                        changes_needed[granttype] = 'remove'
+                        if not dry_run:
+                            statement['Principal']['AWS'].remove(role_arn)
+
+            elif mode == 'deny' and statement['Sid'] == statement_label[granttype] and role_arn in statement['Principal']['AWS']:
+                # we don't selectively deny. that's a grant with a
+                # smaller list. so deny=remove all of this arn.
+                changes_needed[granttype] = 'remove'
+                if not dry_run:
+                    statement['Principal']['AWS'].remove(role_arn)
+
+    if len(changes_needed) and not dry_run:
+        #policy_json_string = json.dumps({'Policy': policy})
+        policy_json_string = json.dumps(policy)
+        kms.put_key_policy(KeyId=keyarn, PolicyName='default', Policy=policy_json_string)
+
+        # returns nothing, so we have to just assume it didn't throw
+        ret['changed'] = True
+
+    ret['changes_needed'] = changes_needed
+    if dry_run:
+        # true if changes > 0
+        ret['changed'] = (not len(changes_needed) == 0)
+
+    return ret
+
+def assert_policy_shape(policy):
+    '''Since the policy seems a little, uh, fragile, make sure we know approximately what we're looking at.'''
+    errors = []
+    if policy['Version'] != "2012-10-17":
+        errors.append('Unknown version/date ({}) of policy. Things are probably different than we assumed they were.'.format(policy['Version']))
+
+    haz = {}
```

Comment:
Not everyone on the codebase is aware that has==haz, or even has English-as-a-first-language. Can you rename this to be clearer?